### PR TITLE
[srpl] implement new peer ID and dataset ID management mechanism

### PR DESCRIPTION
**Pull request by abtink**
This commit contains the following:
- Implement new peer ID assignment mechanism (64-bit ID randomly
  chosen by each peer, new nodes indicate they are "new" in Session
  request message
- Implements dataset ID and Thread Network Data DNS/SRP Anycast
  seq number management model.
- Adds new test-cases:
  - Test to cover dataset ID and seq number mgmnt.
  - Test to cover node removal (re-add and  timeout).
  - Test to cover simultaneous restart of all SRPL peers (power
    outage).

--------

This PR adds a commit on top of commits from  PR:
- https://github.com/openthread/openthread/pull/7257
- We may later squash and combine all the commits into one for merging.

The implementation in this PR follows the ideas/definitions from:
- https://gist.github.com/abtink/4af4fb4f47f439a3b07b617ffe990228

### CODE REVIEW

File: src/core/net/srp_replication.cpp [hunk 1]
```diff
@@ -0,0 +1,2591 @@
...
+
+    txtData.Init(txtDataBuffer, sizeof(txtDataBuffer));
+
+    entries[0].mKey         = kTxtDataKeyDomain;
+    entries[0].mValue       = reinterpret_cast<const uint8_t *>(mDomainName.AsCString());
+    entries[0].mValueLength = StringLength(mDomainName.AsCString(), Dns::Name::kMaxNameSize);
+
+    entries[1].mKey         = kTxtDataKeyAllowsJoin;
+    entries[1].mValue       = kYesString;
+    entries[1].mValueLength = sizeof(kYesString);
```
Comment (superwhd) [inline src/core/net/srp_replication.cpp:544]: Just curious, is there a use case of advertising `join=no`?

Comment (abtink) [reply src/core/net/srp_replication.cpp:544]: I don't think there is any use case for it when SRPL is used in Thread and part of a Thread BR, but perhaps there are use-cases if we have stand-alone SRPL providers.

**Comment by abtink** [comment]
New push adds a new commit:
```
    (to squash) SRPL: adding test API and new test-cases
    
    - test-460-srp-replication-dataset-id-conflict.py
    - test-461-srp-replication-new-peer-with-a-misbehaving-peer.py
    - test-462-srp-replication-peer-id-conflict.py
    - test-463-srp-replication-reconnect-on-failure.py
```

**Comment by abtink** [comment]
Closing this PR in fovor of rebased and updated PR  https://github.com/openthread/openthread/pull/8403.

**Pull request closed without merging**